### PR TITLE
Upload image feature

### PR DESCRIPTION
**Pull request by Dariastep**
The styling is poor, Edit button is responsible for the editing of the name and the email and saving it to the database. The button "upload" is repsponsibe for the uploading of the profile image and saving it to the database.

### CODE REVIEW

File: components/Profile/index.js [hunk 1]
```diff
@@ -40,11 +46,11 @@ export default function Profile({ userProfile }) {
         },
         body: JSON.stringify(requestBody),
       });
-
+      console.log("response1:", response);
```
Comment (actyvystom) [inline components/Profile/index.js]: Please remove `console.logs` before merging.

File: components/Profile/index.js [hunk 2]
```diff
@@ -40,11 +46,11 @@ export default function Profile({ userProfile }) {
...
       if (response.ok) {
         userProfile.name = name; // Update the userProfile object with the new name and email
         userProfile.email = email;
-        /* userProfile.avatar = data.avatar; */
+
         mutate("/api/profile");
         console.log("response is OK");
```
Comment (actyvystom) [inline components/Profile/index.js]: See comment above ⬆️

File: components/Profile/index.js [hunk 3]
```diff
@@ -66,52 +72,81 @@ export default function Profile({ userProfile }) {
...
-  /* 
-  function handleAvatarChange(event) {
-    event.preventDefault();
-    const newAvatar = event.target.files[0];
-    setAvatar(newAvatar);
-  } */
+  async function handleAvatarChange(imageURL) {
+    try {
+      const requestBody = { imageURL: imageURL };
+      console.log(requestBody.imageURL);
```
Comment (actyvystom) [inline components/Profile/index.js]: See comment above ⏫

File: components/Profile/index.js [hunk 4]
```diff
@@ -66,52 +72,81 @@ export default function Profile({ userProfile }) {
...
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(requestBody),
+      });
+
+      if (response.ok) {
+        setAvatar(imageURL);
+        mutate();
+        console.log("Avatar updated in MongoDB:", imageURL);
```
Comment (actyvystom) [inline components/Profile/index.js]: See comments above ⏫

File: components/Profile/index.js [hunk 5]
```diff
@@ -66,52 +72,81 @@ export default function Profile({ userProfile }) {
...
+      } else {
+        console.error(
+          "Unfortunately failed to update avatar in MongoDB",
+          error
+        );
+      }
+    } catch (error) {
+      console.error("Failed to update avatar in MongoDB", error);
```
Comment (actyvystom) [inline components/Profile/index.js:96]: See comments above ⏫

File: components/Profile/index.js [hunk 6]
```diff
@@ -66,52 +72,81 @@ export default function Profile({ userProfile }) {
...
     setEditMode(true); // Turn on editing mode when Edit button is clicked
   }
   return (
-    <ProfileWrapper>
-      <AvatarWrapper>
-        {/*   {editMode ? (
+    <>
+      <ProfileWrapper>
+        <AvatarWrapper>
+          {/*   {editMode ? (
```
Comment (actyvystom) [inline components/Profile/index.js:107]: Please delete commented code.

File: pages/api/images/imagesChange/index.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      const { imageURL } = request.body;
+      if (!imageURL) {
+        throw new Error("Image URL is missing in the request body");
+      }
+
+      const existingUser = await User.findOne();
+      if (existingUser) {
+        existingUser.imageURL = imageURL;
+        await existingUser.save();
+        console.log("Avatar in PUT", imageURL);
```
Comment (actyvystom) [inline pages/api/images/imagesChange/index.js]: See comments on `console.logs` above

File: pages/api/images/imagesChange/index.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+      } else {
+        await User.create({ imageURL });
+      }
+      response.status(200).json({ message: "Avatar updated in MongoDB" });
+    } catch (error) {
+      console.error(error);
```
Comment (actyvystom) [inline pages/api/images/imagesChange/index.js:24]: See comments above ⏫

File: pages/api/images/index.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+      // we use cloudinary search API to retrieve images
+      const result = await cloudinary.v2.search
+        // see documentation to adjust the query at https://cloudinary.com/documentation/search_api#examples
+        .with_field("tags")
+        .max_results(10)
+        .execute();
+      // finally we deliver the response with the result as JSON
+      response.status(200).json(result);
+    } catch (error) {
+      console.error(error);
```
Comment (actyvystom) [inline pages/api/images/index.js:33]: See comments above ⏫

File: pages/api/images/upload/index.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+          } else {
+            // formidable offers us a handy way to get the file data exactly how we need it to pass it to the cloudinary upload API
+            const { file } = files;
+            // newFilename is a temporary filename like e.g. "de49ea6ae2b56c0208f640300"
+            const { newFilename, filepath } = file;
+            // finally we call the cloudinary upload API with our values and pass the temporary filename as public_id
+            const result = await cloudinary.v2.uploader.upload(filepath, {
+              public_id: newFilename
+            });
+            console.log("API: response from cloudinary: ", result);
```
Comment (actyvystom) [inline pages/api/images/upload/index.js]: See comments on `console.logs` above ⏫

File: pages/api/profile/index.js [hunk 11]
```diff
@@ -15,19 +15,20 @@ export default async function handler(request, response) {
     }
   } else if (request.method === "PUT") {
     try {
-      const { name, email } = request.body;
+      const { name, email, avatar } = request.body;
       // Check if there is an existing user
       const existingUser = await User.findOne();
       console.log(existingUser);
```
Comment (actyvystom) [inline pages/api/profile/index.js]: See comments on console.logs above ⏫

File: pages/api/profile/index.js [hunk 12]
```diff
@@ -15,19 +15,20 @@ export default async function handler(request, response) {
...
       if (existingUser) {
         existingUser.name = name;
         existingUser.email = email;
+        existingUser.avatar = avatar;
 
         await existingUser.save();
         console.log(existingUser);
```
Comment (actyvystom) [inline pages/api/profile/index.js]: See comments on console.logs above ⏫

**Review by actyvystom — CHANGES_REQUESTED** [review]
Nice work 🚀 
See some comments on console.logs and commented code lines - and your preview deployment failed, you should check this again.
Keep on hacking 🙂 
Cheers, Thomas

**Review by CorneliusMoe — APPROVED** [review]
LGTM :) 
console.logs cleared✅
deployment works✅

File: components/Profile/index.js (hunk 5 above)
Comment (Dariastep) [reply components/Profile/index.js:96]: I would leave it, cause it is an "important" log in the catch-block :)

File: pages/api/images/imagesChange/index.js (hunk 8 above)
Comment (Dariastep) [reply pages/api/images/imagesChange/index.js:24]: same as here

**Pull request merged**